src/06_ffi.md

```diff
@@ -228,17 +207,316 @@ library offers portable type aliases in `std:os::raw` (or `core::os::raw`):
 The [libc] crate offers more C compatible types that cover almost exhaustively
 the C standard library.
 
-> Rule {{#check FFI-PFTYPE | Use portable aliases `c_*` when binding to platform-dependent types}}
+> ### Rule {{#check FFI-PFTYPE | Use portable aliases `c_*` when binding to platform-dependent types}}
 >
 > In a secure Rust development, when interfacing with foreign code that
 > uses platform-dependent types, such as C's `int` and `long`, Rust code must
 > use portable type aliases, such as provided by the standard library or the
 > [libc] crate, rather than platform-specific types, except if
-> the binding is automatically generated for each platform (for instance,
-> with a [cbindgen] step in the build process).
+> the binding is automatically generated for each platform (see Note below).
+
+<!-- -->
+
+> **Note**
+>
+> Automatic binding generation tools (e.g. [cbindgen], [rust-bindgen]) are able
+> to ensure type consistency on a specific platform. They should be used during
+> the build process for each target to ensure that the generation is sound for
+> the specific target platform.
 
 [libc]: https://crates.io/crates/libc
 
+### Non-robust types: references, pointers, enums
+
+A *trap representation* of a particular type is a representation (pattern of
+bits) that respects the type's representation constraints (such as size and
+alignment) but does not represent a valid value of this type.
```

Comment:
> does not represent a valid value of this type

I think it wouldn't hurt to emphasize that **valid** means that if it is violated at any point in the program _the behavior of the program is no longer defined_ (validity vs safety etc.)

That is, given

```rust
#[no_mangle]
extern "C"
fn foo (_: &'_ i32) {}
```

it is UB to have C call `foo(NULL)`, even if the Rust program does not seem to dereference its input.

Reply:
Yes I think example as in order too!